### PR TITLE
Unittest/flink big query config

### PR DESCRIPTION
**Pull request by Sridhar-Durgam2022**
Unit test cases for FlinkBigQueryConfig class

### CODE REVIEW

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
Please rebase from main so we can merge

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+        instance = null;
+      } finally {
+        instantiated = true;
+      }
+      return instance;
+    }
+
+    public long convertDate(Object object) {
+      final org.joda.time.LocalDate value = (org.joda.time.LocalDate) object;
+      return value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:293]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+      return value.get(DateTimeFieldType.millisOfDay());
+    }
+
+    public long convertTimestamp(Object object) {
+      long dateTime = 0;
+      if (!(object.toString().length() == 26)) {
+        final DateTime value =
+            DateTime.parse(object.toString(), ISODateTimeFormat.dateTimeParser().withZoneUTC());
+
+        dateTime = value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:307]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+      } else if (object.toString().length() == 26) {
+        DateTime value =
+            DateTime.parse(object.toString(), ISODateTimeFormat.dateTimeParser().withZoneUTC());
+        dateTime = value.toDate().getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:311]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:293]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:311]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:307]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:293]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:311]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:307]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java [hunk 4]
```diff
@@ -37,16 +41,45 @@
...
+        MessageSerializer.deserializeSchema(
+                new ReadChannel(
+                    new ByteArrayReadableSeekableByteChannel(
+                        arrowSchemafromReadSession.getSerializedSchema().toByteArray())))
+            .getFields().stream()
+            .forEach(
+                field -> {
+                  readSessionFieldNames.add(field.getName());
+                });
+        arrowSchema =
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java]: addressed

File: pom.xml [hunk 5]
```diff
@@ -236,6 +237,16 @@
             <artifactId>jsqlparser</artifactId>
             <version>${jsqlparser.version}</version>
         </dependency>
+        <dependency>
+           <groupId>joda-time</groupId>
+           <artifactId>joda-time</artifactId>
+           <version>${joda.version}</version>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply pom.xml:246]: addressed

File: src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/ArrowRowDataDeserializationSchema.java]: addressed

File: src/main/java/com/google/cloud/flink/bigquery/BigQuerySinkFunction.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+  public BigQuerySinkFunction(
+      org.apache.flink.table.api.Table sourceResultTable,
+      String projectId,
+      String dataset,
+      String table)
+      throws JSQLParserException {
+    BigQueryDirectDataWriterContext writeContextObj =
+        new BigQueryDirectDataWriterContext(sourceResultTable, projectId, dataset, table);
+    writeContext = writeContextObj;
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/BigQuerySinkFunction.java]: addressed

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:311]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:293]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/main/java/com/google/cloud/flink/bigquery/FlinkBigQueryConfig.java [hunk 7]
```diff
@@ -106,9 +106,9 @@ public static WriteMethod from(@Nullable String writeMethod) {
...
 
   static final String GCS_CONFIG_CREDENTIALS_FILE_PROPERTY =
-      "google.cloud.auth.service.account.json.keyfile";
+      System.getenv("GOOGLE_APPLICATION_CREDENTIALS");
+  // TODO: Read projectId from json filepath
   static final String GCS_CONFIG_PROJECT_ID_PROPERTY = "q-gcp-6750-pso-gs-flink-22-01";
-  private static final String READ_DATA_FORMAT_OPTION = DataFormat.ARROW.toString();
   private static final ImmutableList<String> PERMITTED_READ_DATA_FORMATS =
       ImmutableList.of(DataFormat.ARROW.toString(), DataFormat.AVRO.toString());
   private static final Supplier<com.google.common.base.Optional<String>> DEFAULT_FALLBACK =
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/FlinkBigQueryConfig.java]: @davidrabinowitz , We can discuss on this as Spark connector using the same

File: src/test/java/com/google/cloud/flink/bigquery/FlinkBQConfigTest.java [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+    assertThat(config.getDefaultParallelism()).isEqualTo(10);
+    assertThat(config.getMaterializationDataset()).isEqualTo(Optional.ofNullable("vmd"));
+    assertThat(config.getMaterializationExpirationTimeInMinutes()).isEqualTo(1440);
+    assertThat(config.getMaterializationProject()).isEqualTo(Optional.ofNullable("vmp"));
+    assertThat(config.getSelectedFields()).isEqualTo("word,word_count");
+  }
+
+  @Ignore
+  @Test
+  public void testInvalidCompressionCodec() {
```
Comment (VinayLondhe14) [inline src/test/java/com/google/cloud/flink/bigquery/FlinkBQConfigTest.java:189]: Why do we have this ignored test here? Let us please remove it

Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/FlinkBQConfigTest.java:187]: Removed

**Review by VinayLondhe14 — APPROVED** [review]

**Pull request closed without merging**